src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionItemExtensions.cs

```diff
@@ -60,15 +65,33 @@ public static VSCompletionItem ToVSCompletionItem(this CompletionItem completion
                 Detail = completion.Detail,
                 Documentation = completion.Documentation,
                 FilterText = completion.FilterText,
-                InsertText = completion.FilterText,
+                InsertText = completion.InsertText,
                 InsertTextFormat = completion.InsertTextFormat,
                 Kind = completion.Kind,
                 Label = completion.Label,
                 Preselect = completion.Preselect,
                 SortText = completion.SortText,
                 Tags = completion.Tags,
                 TextEdit = completion.TextEdit,
+                VsCommitCharacters = vsCommitCharacters,
             };
+
+            static IEnumerable<VSCommitCharacter> GetVSCommitCharacters(CompletionItem completion)
```

Comment:
Do we need to gate this on the VS LSP++ capability or is this method only ever called if we know we're in a VS context? 

For context [this](https://github.com/dotnet/razor-tooling/blob/340ee1728c3d0e4eca0a49681c113362c9e86ad7/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs#L201) is an example of where we decide to use more colorful descriptions in VS scenarios based on the capability

Reply:
Only called when we support VSCommitCharacters.